aten/src/TH/generic/THTensorLapack.cpp

```diff
@@ -121,9 +121,8 @@ void THTensor_(gels)(THTensor *rb_, THTensor *ra_, THTensor *b, THTensor *a)
   TORCH_CHECK(a->size(0) == b->size(0), "Expected A and b to have same size "
       "at dim 0, but A has ", a->size(0), " rows and B has ", b->size(0), " rows");
 
-  if (THTensor_nDimensionLegacyAll(b) == 1) {
-    b = THTensor_(newWithStorage2d)(THTensor_getStoragePtr(b), b->storage_offset(), b->size(0),
-            b->stride(0), 1, 0);
+  if (b->dim() == 1) {
+    b = THTensor_wrap(b).unsqueeze(1).unsafeReleaseTensorImpl();
```

Comment:
Don't know nearly enough to check whether this change is semantics-preserving. Should I take it on faith or go look at stuff?

Reply:
good question.  There are two changes:
1) the `b->dim() == 1` change is semantics preserving because the LegacyAll stuff maps between old TH-style sizes (i.e. 0-dim scalars are 1-D, numel == 0 tensors are 0-d and current torch semantics).  This was just a codemod we did at some point to support numpy-like sizes without changing any semantics.  But if you look above you see we are already checking that the dimensionality of B is 1 or 2 -- so this check is equivalent to checking that the dimensionality of B is 1.

Reply:
good question.  There are two changes:
1) the `b->dim() == 1` change is semantics preserving because the LegacyAll stuff maps between old TH-style sizes (i.e. 0-dim scalars are 1-D, numel == 0 tensors are 0-d and current torch semantics).  This was just a codemod we did at some point to support numpy-like sizes without changing any semantics.  But if you look above you see we are already checking that the dimensionality of B is 1 or 2 -- so this check is equivalent to checking that the dimensionality of B is 1.

Reply:
for the `newWithStorage`, that's saying use the same tensor but make the size `{b->size(0), 1}` and the stride `{b->stride(0), 0)`.  That's just a view, although overconstrained: the `0` stride can be anything, and the unsqueeze in this case sets it to 1.